File: Api/rest_assured/src/test/java/support/utils/RestContext.java
package support.utils;

import io.restassured.RestAssured;
import io.restassured.response.Response;
import io.restassured.specification.RequestSpecification;

import java.util.Map;

public class RestContext {
    private static RequestSpecification request;
    private static Response response;

    public static void initRequest(){
        RestAssured.useRelaxedHTTPSValidation();
        request = RestAssured.given();
    }

    public static void log(){
        RestAssured.given().then().log().all();
    }

    public static void setPath(String baseUri, String path){
        if ( request == null){
            initRequest();
        }
        request.baseUri(baseUri);
        request.basePath(path);
        RestAssured.useRelaxedHTTPSValidation();
    }

    public static void setHeader(Map<String, String> contentHeader){
       request.headers(contentHeader);
    }

    public static void setDefaultHeader(){

    }

    public static void setBody(String contentBody){
        request.body(contentBody);

    }

    public static void getRequest(){
        response = request.get();

    }

    public static void postRequest(){
        response = request.post();

    }

    public static void setPathParams(Map<String, String> params){
        request.pathParams(params);

    }

    public static void setParams(Map<String, ?> params){
        request.params(params);

    }

    public static void putRequest(){
       response = request.put();

    }

    public static void patchRequest(){
       response = request.patch();

    }

    public static Response getResponse(){
       return response;
    }
}

File: Api/rest_assured_kotlin/src/test/java/support/utils/RestContext.kt
package support.utils

import io.restassured.specification.RequestSpecification
import io.restassured.RestAssured
import io.restassured.response.Response

object RestContext {
    private var request: RequestSpecification? = null
    @JvmStatic
    var response: Response? = null
        private set

    fun initRequest() {
        RestAssured.useRelaxedHTTPSValidation()
        request = RestAssured.given()
    }

    fun log() {
        RestAssured.given().then().log().all()
    }

    fun setPath(baseUri: String?, path: String?) {
        if (request == null) {
            initRequest()
        }
        request!!.baseUri(baseUri)
        request!!.basePath(path)
        RestAssured.useRelaxedHTTPSValidation()
    }

    fun setHeader(contentHeader: MutableMap<String, String>) {
        request!!.headers(contentHeader)
    }

    fun setDefaultHeader() {}
    fun setBody(contentBody: String?) {
        request!!.body(contentBody)
    }

    fun getRequest() {
        response = request!!.get()
    }

    fun postRequest() {
        response = request!!.post()
    }

    fun setPathParams(params: Map<String?, String?>?) {
        request!!.pathParams(params)
    }

    fun setParams(params: Map<String?, *>?) {
        request!!.params(params)
    }

    fun putRequest() {
        response = request!!.put()
    }

    fun patchRequest() {
        response = request!!.patch()
    }
}